opentelemetry-sdk/tests/logs/test_export.py

```diff
@@ -463,161 +468,129 @@ def test_validation_negative_max_queue_size(self):
             max_export_batch_size=101,
         )
 
-    def test_shutdown(self):
-        exporter = InMemoryLogExporter()
-        log_record_processor = BatchLogRecordProcessor(exporter)
-
-        provider = LoggerProvider()
-        provider.add_log_record_processor(log_record_processor)
-
-        logger = logging.getLogger("shutdown")
-        logger.addHandler(LoggingHandler(logger_provider=provider))
-
-        with self.assertLogs(level=logging.WARNING):
-            logger.warning("warning message: %s", "possible upcoming heatwave")
-        with self.assertLogs(level=logging.WARNING):
-            logger.error("Very high rise in temperatures across the globe")
-        with self.assertLogs(level=logging.WARNING):
-            logger.critical("Temperature hits high 420 C in Hyderabad")
+    def test_logs_exported_once_batch_size_reached(self):
+        exporter = Mock()
+        log_record_processor = BatchLogRecordProcessor(
+            exporter=exporter,
+            max_queue_size=15,
+            max_export_batch_size=15,
+            # Will not reach this during the test, this sleep should be interrupted when batch size is reached.
+            schedule_delay_millis=30000,
+        )
+        before_export = time.time_ns()
+        for _ in range(15):
+            log_record_processor.emit(EMPTY_LOG)
+        # Wait a bit for the worker thread to wake up and call export.
+        time.sleep(0.1)
+        exporter.export.assert_called_once()
+        after_export = time.time_ns()
+        # Shows the worker's 30 second sleep was interrupted within a second.
+        self.assertTrue((after_export - before_export) < 1e9)
+
+    # pylint: disable=no-self-use
+    def test_logs_exported_once_schedule_delay_reached(self):
+        exporter = Mock()
+        log_record_processor = BatchLogRecordProcessor(
+            exporter=exporter,
+            # Should not reach this during the test, instead export should be called when delay millis is hit.
+            max_queue_size=15,
+            max_export_batch_size=15,
+            schedule_delay_millis=100,
+        )
+        for _ in range(15):
+            log_record_processor.emit(EMPTY_LOG)
+            time.sleep(0.11)
+        exporter.export.assert_has_calls(
+            [call([EMPTY_LOG]) for _ in range(15)]
+        )
```

Comment:
If it's just testing the scheduled export, why not just emit a single log for the test?